nautobot/vpn/models.py

```diff
@@ -516,12 +515,22 @@ def clean(self):
             raise ValidationError("Source IP Address and Source FQDN are mutually exclusive fields. Select only one.")
         if not any([self.source_interface, self.source_ipaddress, self.source_fqdn]):
             raise ValidationError("Source Interface or Source IP Address or Source FQDN Is required.")
-        if self.source_interface and not self.source_interface.device:
+        if (
+            self.source_interface
+            and not self.source_interface.parent
+            and (not hasattr(self.source_interface.module, "device") or not self.source_interface.module.device)
+        ):
             raise ValidationError("Source Interface must belong to a device.")
+        if (
+            self.source_ipaddress
+            and self.source_interface
+            and (self.source_ipaddress not in self.source_interface.ip_addresses.all())
+        ):
+            raise ValidationError("Source IP address must be assigned to Source Interface.")
         return super().clean()
 
     def save(self, *args, **kwargs):
-        if self.source_interface and not self.device:
-            self.device = self.source_interface.device
+        if self.source_interface:
```

Comment:
Explanation: because on update if an interface is changed to different device/intf then device field should be updated.